### ISSUE TITLE
RCS1102 triggers false positive when inherited by nested class

### ISSUE THREAD

**Issue by gtbuchanan** [JosefPihrt/Roslynator#977]
**Product and Version Used**: 4.1.1

**Steps to Reproduce**:
1. Create a generic class
2. Add a nested class (RCS1102 is reported here)
3. Make the inner class inherit the outer class (RCS1102 is still reported)

This strategy is used in [certain unit testing structures](https://haacked.com/archive/2012/01/02/structuring-unit-tests.aspx/) where there may be no instance members/methods. For example, passing a type argument to `TestFixtureAttribute` from NUnit for a generic test fixture requires that the nested class inherit from the child class in the aforementioned structure.

**Actual Behavior**:
```csharp
public class Class1<T> // RCS1102 is reported on Class1
{
    public sealed class Class2 : Class1<T>
    {
    }
}
```

**Expected Behavior**:
[RCS1102](https://github.com/JosefPihrt/Roslynator/blob/main/docs/analyzers/RCS1102.md) should not be reported since it is being inherited by the nested class. Making `Class1` static in this case would result in a compilation error due to the inheritance.
